pkg/controller/controller_instance.go

```diff
@@ -339,6 +443,18 @@ func (c *controller) reconcileServiceInstanceAdd(instance *v1beta1.ServiceInstan
 		return nil
 	}
 
+	// don't DOS the broker.  If we already did a provision attempt that ended with a non-terminal
+	// error then we set a next retry time.
+	if delay := c.getDelayForProvisionRetry(instance); delay > 0 {
```

Comment:
What if the instance spec was updated? have we purged the item from the `provisionRetryQueue` by this time?
We should reset the backoff when we start a new operation with generation incremented?

Reply:
good point.  I've updated the map to use the instance with the generation appended now.  This ensures if the instance is updated the backoff will be reset.